ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/StringInterningCacheSpec.scala

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.interning
+
+import com.daml.lf.data.Ref
+import com.daml.logging.LoggingContext
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+import scala.util.Try
+
+class StringInterningCacheSpec extends AsyncFlatSpec with Matchers {
+  private implicit val lc: LoggingContext = LoggingContext.ForTesting
+
+  behavior of "PrefixingStringInterningCache"
+
+  it should "provide working cache by extending" in {
+    val testee = new PrefixingStringInterningCache((_, _) => _ => Future.successful(Nil))
+    partyShouldNotPresent(testee, "p1")
+    partyShouldNotPresent(testee, "p2")
+    partyShouldNotPresent(testee, "22:same:name")
+    templateShouldNotPresent(testee, "22:t:a")
+    templateShouldNotPresent(testee, "22:t:b")
+    templateShouldNotPresent(testee, "22:same:name")
+    testee.add(
+      new DomainStringIterators(
+        parties = List("p1", "p2", "22:same:name").iterator,
+        templateIds = List("22:t:a", "22:t:b", "22:same:name").iterator,
+      )
+    ) shouldBe Vector(
+      1 -> "p|p1",
+      2 -> "p|p2",
+      3 -> "p|22:same:name",
+      4 -> "t|22:t:a",
+      5 -> "t|22:t:b",
+      6 -> "t|22:same:name",
+    )
+    partyShouldPresent(testee, "p1", 1)
+    partyShouldPresent(testee, "p2", 2)
+    partyShouldPresent(testee, "22:same:name", 3)
+    partyShouldNotPresent(testee, "unknown")
+    templateShouldPresent(testee, "22:t:a", 4)
```

Comment:
Could you rename this asserting functions not to be so similar when skimming through tests? E.g.:
```
partyShouldPresent -> assertPartyPresent
partyShouldNotPresent -> assertPartyAbsent
```

or even:

```
partyShouldPresent -> partyPresent
partyShouldNotPresent ->partyAbsent
```

Reply:
good idea